lib/Doctrine/Migrations/Tools/Console/ConsoleInputMigratorConfigurationFactory.php

```diff
@@ -31,12 +32,26 @@ public function getMigratorConfiguration(InputInterface $input): MigratorConfigu
 
     private function determineAllOrNothingValueFrom(InputInterface $input): ?bool
     {
-        $allOrNothingOption = null;
+        $allOrNothingOption        = null;
+        $wasOptionExplicitlyPassed = $input->hasOption('all-or-nothing');
 
-        if ($input->hasOption('all-or-nothing')) {
+        if ($wasOptionExplicitlyPassed) {
             $allOrNothingOption = $input->getOption('all-or-nothing');
         }
 
+        if ($wasOptionExplicitlyPassed && $allOrNothingOption !== null) {
+            Deprecation::trigger(
+                'doctrine/migrations',
+                'https://github.com/doctrine/migrations/issues/1304',
+                <<<'DEPRECATION'
+                    Context: Passing values to the `--all-or-nothing`
+                    Problem: The mechanism to read the option value is unnecessarily complicated.
+                    Solution: From version 4.0.x, `--all-or-nothing` option won't accept any value, 
```

Comment:
The solution should describe how to avoid the deprecation.

```suggestion
                    Solution: If you need to disable the behavior, omit the option,
                    otherwise, pass the option without a value
```

Reply:
> If you need to disable the behavior, omit the option,

From my understanding, this won't be the case. If the configuration has this option set to `true`, omitting the CLI option does not change it to `false`, or is this what is planned for the 4.x version?

Reply:
I updated the Deprecation with your suggestion.